File: comparisons/categorical_anova.R
suppressPackageStartupMessages(library(lme4))
suppressPackageStartupMessages(library(lmerTest))
suppressPackageStartupMessages(library(car))

cat("\n=== R (lme4 + lmerTest) ===\n")
data <- read.csv("tests/data/pastes.csv")
data$cask <- as.factor(data$cask)
data$batch <- as.factor(data$batch)

fit <- lmer(strength ~ cask + (1 | batch), data = data, REML = TRUE)
print(summary(fit))

cat("\n=== Type III ANOVA (Satterthwaite) ===\n")
print(anova(fit, ddf="Satterthwaite", type=3))

cat("\n=== Type III ANOVA (Wald Chi-Square approximation for joint) ===\n")
print(car::Anova(fit, type=3, test.statistic="Wald"))

File: comparisons/categorical_anova.rs
use lme_rs::{lmer, DdfMethod};
use polars::prelude::*;
use std::fs::File;

fn main() -> Result<(), Box<dyn std::error::Error>> {
    println!("\n=== Rust (lme-rs) ===");
    let mut file = File::open("tests/data/pastes.csv")?;
    let df = CsvReadOptions::default()
        .with_has_header(true)
        .into_reader_with_file_handle(&mut file)
        .finish()?;

    // Evaluate lmer and Satterthwaite implicitly
    let mut fit = lmer("strength ~ cask + (1 | batch)", &df, true)?;
    fit.with_satterthwaite(&df)?;

    println!("{}", fit);

    println!("\n=== Type III ANOVA (Satterthwaite 1-DoF / Joint Wald) ===");
    let anova_res = fit.anova(DdfMethod::Satterthwaite)?;
    println!("{}", anova_res);

    Ok(())
}
